fairlearn/reductions/_moments/conditional_selection_rate.py

```diff
@@ -36,12 +38,22 @@ class ConditionalSelectionRate(ClassificationMoment):
     where error(A = a) = total_error
     """
 
-    def __init__(self, ratio=1.0):
+    def __init__(self, *, difference_bound=None, ratio_bound=None, ratio_bound_slack=0.0):
         """Initialize with the ratio value."""
         super(ConditionalSelectionRate, self).__init__()
-        if not (0 < ratio <= 1):
-            raise ValueError(_MESSAGE_RATIO_NOT_IN_RANGE)
-        self.ratio = ratio
+        if (difference_bound is None) and (ratio_bound is None):
+            self.eps = _DEFAULT_DIFFERENCE_BOUND
+            self.ratio = 1.0
+        if difference_bound is not None and ratio_bound is not None:
```

Comment:
I think that we should rewrite the case analysis using `elif` to make it clearer that only one of the cases is going to happen. Right now, I need to manually verify such a flow. I suggest the following sequence:
```python
if (difference_bound is None) and (ratio_bound is None):
    ...
elif (difference_bound is not None) and (ratio_bound is None):
    ...
elif (difference_bound is None) and (ratio_bound is not None):
    ...
else: # both difference_bound and ratio_bound are not None
    ...
```